app/controllers/users.js

```diff
@@ -152,37 +142,52 @@ exports.addDonation = function(req, res) {
   res.send();
 };
 
-/**
- *  Show profile
- */
-exports.show = function(req, res) {
-  var user = req.profile;
+// Show user profile
+exports.show = (req, res) => {
+  const user = req.profile;
 
   res.render('users/show', {
     title: user.name,
-    user: user
+    user
   });
 };
 
-/**
- * Send User
- */
-exports.me = function(req, res) {
+// Send User
+exports.me = (req, res) => {
   res.jsonp(req.user || null);
 };
 
-/**
- * Find user by id
- */
-exports.user = function(req, res, next, id) {
+// Find user by id
+exports.user = (req, res, next, id) => {
   User
     .findOne({
       _id: id
     })
-    .exec(function(err, user) {
+    .exec((err, user) => {
       if (err) return next(err);
-      if (!user) return next(new Error('Failed to load User ' + id));
+      if (!user) return next(new Error(`Failed to load User ${id}`));
       req.profile = user;
       next();
     });
-};
\ No newline at end of file
+};
+
+ // Attach token to user credentials after authentication
+exports.loginWithEmail = (req, res) => {
+  User
+    .findOne({ email: req.body.email })
```

Comment:
There should be better error handling here. What if email or password are empty. We should not respond with `user not found`. Or try to authenticate an empty pasword